sormas-ui/src/main/java/de/symeda/sormas/ui/events/eventParticipantMerge/PickLeadEventParticipant.java

```diff
@@ -0,0 +1,158 @@
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.VaadinUiUtil;
+
+public class PickLeadEventParticipant extends CustomField<List<String>> {
+
+	private Map<String, EventParticipantGridByEvent> duplicateEventParticipantByEventGridMap;
+
+	protected VerticalLayout mainLayout;
+
+	public PickLeadEventParticipant(List<EventParticipantSelectionDto> eventParticipantSelectionDtos) {
+		Map<EventReferenceDto, List<EventParticipantSelectionDto>> eventParticipantsByEventSelectionDtos =
+			eventParticipantSelectionDtos.stream().collect(Collectors.groupingBy(EventParticipantSelectionDto::getEvent));
+
+		duplicateEventParticipantByEventGridMap = initializeDuplicateEventParticipantsByEventGrids(eventParticipantsByEventSelectionDtos);
+	}
+
+	@Override
+	protected Component initContent() {
+		mainLayout = new VerticalLayout();
+		mainLayout.setSpacing(true);
+		mainLayout.setMargin(false);
+		mainLayout.setSizeUndefined();
+		mainLayout.setWidth(100, Unit.PERCENTAGE);
+
+		HorizontalLayout infoLayout = new HorizontalLayout();
+		infoLayout.addComponent(VaadinUiUtil.createInfoComponent(I18nProperties.getString(Strings.infoPickEventParticipantsForPersonMerge)));
+		mainLayout.addComponent(infoLayout);
+
+		duplicateEventParticipantByEventGridMap.forEach((key, value) -> {
+			mainLayout.addComponent(value.getEventTitle());
+			mainLayout.addComponent(value.eventParticipantMergeSelectionGrid);
+		});
+
+		return mainLayout;
+	}
+
+	@Override
+	public Class<? extends List<String>> getType() {
+		return (Class<? extends List<String>>) new ArrayList<String>(0).getClass();
+	}
+
+	@Override
+	public List<String> getValue() {
+		return duplicateEventParticipantByEventGridMap.values().stream().map(eventParticipantGridByEvent -> {
+			Set<EventParticipantSelectionDto> selectedRow = eventParticipantGridByEvent.eventParticipantMergeSelectionGrid.getSelectedItems();
+			if (!selectedRow.isEmpty()) {
+				return selectedRow.iterator().next().getUuid();
+			}
+			return null;
+		}).filter(Objects::nonNull).collect(Collectors.toList());
+	}
+
+	protected Map<String, EventParticipantGridByEvent> initializeDuplicateEventParticipantsByEventGrids(
+		Map<EventReferenceDto, List<EventParticipantSelectionDto>> eventParticipantByEventDtos) {
+
+		Map<String, EventParticipantGridByEvent> ret = new HashMap<>();
+
+		eventParticipantByEventDtos.forEach((key, value) -> {
+
+			EventParticipantMergeSelectionGrid newGrid = new EventParticipantMergeSelectionGrid(value);
+			Component eventComponent = buildEventComponent(key.getUuid(), key.getCaption());
+
+			EventParticipantGridByEvent eventParticipantGridByEvent = new EventParticipantGridByEvent();
+			eventParticipantGridByEvent.setEventTitle(eventComponent);
+			eventParticipantGridByEvent.setEventParticipantMergeSelectionGrid(newGrid);
+
+			ret.put(key.getUuid(), eventParticipantGridByEvent);
+
+		});
+		return ret;
+	}
+
+	private static class EventParticipantGridByEvent {
```

Comment:
Please rename to `EventParticipantSelectGrid`

Reply:
renamed